cmd/cnab-to-oci/fixup.go

```diff
@@ -54,12 +55,23 @@ func runFixup(opts fixupOptions) error {
 	if err := json.Unmarshal(bundleJSON, &b); err != nil {
 		return err
 	}
-	resolver := createResolver(opts.insecureRegistries)
+	resolver, blobsource := createResolver(opts.insecureRegistries)
 	ref, err := reference.ParseNormalizedNamed(opts.targetRef)
 	if err != nil {
 		return err
 	}
-	if err := remotes.FixupBundle(context.Background(), &b, ref, resolver); err != nil {
+	eventsCh := make(chan remotes.FixupEvent)
+	eventsShown := make(chan struct{})
+	go func() {
+		defer close(eventsShown)
+		for ev := range eventsCh {
+			displayEvent(os.Stderr, ev)
+		}
+	}()
+	err = remotes.FixupBundle(context.Background(), &b, ref, resolver, blobsource, eventsCh)
```

Comment:
I think we should avoid exposing channels in our APIs, it should be purely internal.

Reply:
I think this case is justifiable -- it allows us to have a library doing something complex and have clients consume an event stream which they render according to their local UI.

I think of this as logically an "event stream" rather than a "channel" (which is an implementation detail really).

FWIW buildkit has the same model, I think it's a pretty common mechanism for long running async library calls in Go.